pkg/vpcmodel/explainabilityConnectivity.go

```diff
@@ -135,79 +135,79 @@ func (c *VPCConfig) computeExplainRules(srcNodes, dstNodes []Node,
+			computeActualRulesGivenRulesFilter(&singleSrcDstDetails.potentialAllowRules.ingressRules, filterRelevant)
+		actualAllowEgress, egressEnabled :=
+			computeActualRulesGivenRulesFilter(&singleSrcDstDetails.potentialAllowRules.egressRules, filterRelevant)
+		actualDenyIngress, _ := computeActualRulesGivenRulesFilter(&singleSrcDstDetails.potentialDenyRules.ingressRules, filterRelevant)
+		actualDenyEgress, _ := computeActualRulesGivenRulesFilter(&singleSrcDstDetails.potentialDenyRules.egressRules, filterRelevant)
 		actualAllow := &rulesConnection{*actualAllowIngress, *actualAllowEgress}
 		actualDeny := &rulesConnection{*actualDenyIngress, *actualDenyEgress}
 		singleSrcDstDetails.actualAllowRules = actualAllow
 		singleSrcDstDetails.ingressEnabled = ingressEnabled
 		singleSrcDstDetails.egressEnabled = egressEnabled
 		singleSrcDstDetails.actualDenyRules = actualDeny
 	}
-	return nil
 }
 
-// computes actual rules relevant to the connection, as well as whether the direction is enabled
-func computeActualRules(rulesLayer *rulesInLayers, filtersExternal map[string]bool, srcDstInternal bool) (*rulesInLayers, bool) {
+// given rulesInLayers and a filter computes actual rules and whether the direction is enabled.
+// this is called separately for each direction (ingreee/egress) and allow/deny
+func computeActualRulesGivenRulesFilter(rulesLayers *rulesInLayers, filters map[string]bool) (*rulesInLayers, bool) {
 	actualRules := rulesInLayers{}
-	filterNotBlocking := map[string]bool{}
-	for filter, potentialRules := range *rulesLayer {
-		filterIsRelevant := filtersExternal[filter] || srcDstInternal
-		if filterIsRelevant {
-			actualRules[filter] = potentialRules
-		}
-		// The filter is not blocking if it has enabling  rules or is not required for the router
-		// Specifically, current filters are nacl and sg; if both src and dst are internal then they are both relevant.
-		// (if both are in the same subnet then the nacl analyzer will handle it correctly.)
-		// If fip is the router and one of src/dst is external then nacl is ignored.
-		if filterHasRelevantRules(potentialRules) || !filterIsRelevant {
-			// The case of two vsis of the same subnet is tricky: the nacl filter is relevant but there are no potential rules
-			// this is solved by adding a dummy rule for this case with index -1 (DummyRule),
-			// so that potentialRules here will not be empty
-			// the printing functionality ignores rules of index -1 (DummyRule)
-			// thus nacl will not be identified as a blocking filter in this case
-			filterNotBlocking[filter] = true
-		}
-	}
 	directionEnabled := true
-	for _, filter := range filterLayers {
-		if !filterNotBlocking[filter] {
+	for _, layer := range filterLayers {
+		filterIsRelevant := filters[layer]
+		potentialRules := (*rulesLayers)[layer]
+		// The filter blocking if it is relevant or has no allow rules
+		if filterIsRelevant && !filterHasRelevantRules(potentialRules) {
 			directionEnabled = false
```

Comment:
no need to preserve `directionEnabled` per layer?

Reply:
The explainability in this context includes (at the moment)
* Whether the entire direction (ingress/egress) is enabled
* For each relevant table - is it blocking/allowing

So no, `directionEnabled`  per layer is not required